### PR TITLE
Yaml hyperparams

### PR DESCRIPTION
**Pull request by wiktorfilipiuk**

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -183,30 +190,31 @@ def main(args):
...
+                           transformer_name="sign_language_transformer",
+                           num_attention_heads=model_config['transformer']['num_attention_heads'],
+                           transformer_dropout_rate=model_config['transformer']["dropout_rate"],
+                           num_segments=args.num_segments,
+                           model_save_dir=args.save,
+                           neptune=args.neptune,
+                           representation_size=model_config['feature_extractor']["representation_size"],
+                           feedforward_size=model_config['transformer']["feedforward_size"],
+                           num_encoder_layers=model_config['transformer']["num_encoder_layers"],
+                           transformer_output_size=model_config['transformer']["output_size"],
```
Comment (alicjak1519) [inline train.py:215]: Let's unify the quotation marks :)

**Review by alicjak1519 — COMMENTED** [review]
It looks nice and useful! :)

File: models/model.py [hunk 2]
```diff
@@ -31,42 +31,57 @@ class GlossTranslationModel(pl.LightningModule):
...
+                "feedforward_size": feedforward_size,
+                "num_encoder_layers": num_encoder_layers,
+                "num_segments": num_segments,
+                "num_attention_heads": num_attention_heads,
+                "transformer_dropout_rate": transformer_dropout_rate,
+                "classification_mode": classification_mode,
+                "feature_extractor_name": feature_extractor_name,
+                "feature_extractor_model_path": feature_extractor_model_path,
+                "transformer_name": transformer_name,
+            }
```
[the remark this answers is not in the document]
Comment (wiktorfilipiuk) [reply models/model.py:71]: @AgaMiko @majsylw What do you think about it? I think it's a reasonable idea.

File: train_config_default.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    model_path: "efficientnet_b1"
+    representation_size: 512
+
+transformer:
+    feedforward_size: 1024
+    transformer_output_size: 784
+    num_attention_heads: 4
+    num_encoder_layers: 2
+    output_size: 1024
+    dropout_rate: 0.1
```
[the remark this answers is not in the document]
Comment (wiktorfilipiuk) [reply train_config_default.yml:11]: @AgaMiko  @majsylw What do you think about it? I think it's a reasonable idea.

File: models/model.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wiktorfilipiuk) [reply models/model.py:71]: As for now all information that is in the train_config.yml file is being logged into neptune's metadata

File: train_config_default.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (AgaMiko) [reply train_config_default.yml:11]: I like it!

**Review by AgaMiko — APPROVED** [review]

File: train_config_default.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (majsylw) [reply train_config_default.yml:11]: great, what about parameters for classification heads? The config also need to be given in the json file

**Review by mariaogryczak — APPROVED** [review]

File: train_config_default.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (wiktorfilipiuk) [reply train_config_default.yml:11]: @majsylw Let's discuss it on Tuesday. Maybe there are more things we would like to add to the config and other parties would be better suited for the implementation :)

[the remark this answers is not in the document]
Comment (wiktorfilipiuk) [reply train_config_default.yml:11]: For now, I will merge this PR as it is and add other parameters mentioned by @alicjak1519 in next PR.

**Pull request merged**